pkg/webhook/work/mutating.go

```diff
@@ -97,5 +97,9 @@ func removeIrrelevantField(workload *unstructured.Unstructured) {
 
 	unstructured.RemoveNestedField(workload.Object, "status")
 
-	unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
+	// keep clusterIP filed when clusterIP = None
+	clusterIP, exist, _ := unstructured.NestedString(workload.Object, "spec", "clusterIP")
+	if clusterIP != "None" && exist {
+		unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
+	}
```

Comment:
Thanks. I think it would be better to check the resource type. like:

```golang
	if workload.GetKind() == util.ServiceKind {
		// In the case spec.clusterIP is set to `None`, means user want a headless service,  then it shouldn't be removed.
		clusterIP, exist, _ := unstructured.NestedString(workload.Object, "spec", "clusterIP")
		if exist && clusterIP != corev1.ClusterIPNone {
			unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
		}
	}
```

Reply:
Just post the whole diff:
```diff
diff --git a/pkg/webhook/work/mutating.go b/pkg/webhook/work/mutating.go
index f7cb8c7..ea6b0f6 100644
--- a/pkg/webhook/work/mutating.go
+++ b/pkg/webhook/work/mutating.go
@@ -5,12 +5,14 @@ import (
        "encoding/json"
        "net/http"

+       corev1 "k8s.io/api/core/v1"
        "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
        "k8s.io/apimachinery/pkg/runtime"
        "k8s.io/klog/v2"
        "sigs.k8s.io/controller-runtime/pkg/webhook/admission"

        workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+       "github.com/karmada-io/karmada/pkg/util"
 )

 // MutatingAdmission mutates API request if necessary.
@@ -97,9 +99,11 @@ func removeIrrelevantField(workload *unstructured.Unstructured) {

        unstructured.RemoveNestedField(workload.Object, "status")

-       // keep clusterIP filed when clusterIP = None
-       clusterIP, exist, _ := unstructured.NestedString(workload.Object, "spec", "clusterIP")
-       if clusterIP != "None" && exist {
-               unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
+       if workload.GetKind() == util.ServiceKind {
+               // In the case spec.clusterIP is set to `None`, means user want a headless service,  then it shouldn't be removed.
+               clusterIP, exist, _ := unstructured.NestedString(workload.Object, "spec", "clusterIP")
+               if exist && clusterIP != corev1.ClusterIPNone {
+                       unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
+               }
        }
 }
```